lib/charms/kubernetes_charm_libraries/v0/multus.py

```diff
@@ -537,8 +540,16 @@ def _configure_network_attachment_definitions(self):
             list of NetworkAttachmentDefinitions to create
           - Else, delete it
         2. Goes through the list of NetworkAttachmentDefinitions to create and create them all
+        3. Detects the NAD config changes and triggers pod restart
+           if any there is any modification in existing NADs
         """
-        network_attachment_definitions_to_create = self.network_attachment_definitions_func()
+        if (
+            network_attachment_definitions_to_create := self.network_attachment_definitions_func()
+        ) is None:
+            # If the NAD configs are not valid, this is set to None
+            # by charm to exit without processing as a protection.
```

Comment:
I'm curious what the implementation looks like on the charm side for this.

It seems like invalid NAD configs are something the charm should be aware of, and avoid making it here in the first place. I would expect the library to throw exceptions if something is wrong, but the charm to be responsible for making sure that doesn't happen.

Perhaps my intuition is wrong considering the charm lifecycle, though.

Reply:
Hello Daniel,
You can check the implementation in the charm part as follows. It will be set to Blocked status.
https://github.com/canonical/sdcore-upf-operator/blob/3cf3b0f3a9555f48373f637999355d057b09e7e0/src/charm.py#L176

Reply:
Resolving based on discussion today in daily.